packages/database/src/modelClasses/DataSource.js

```diff
@@ -127,25 +122,22 @@ export class DataSourceModel extends DatabaseModel {
     });
   }
 
-  // TODO: Remove this function and the code directly associated with it when
-  // https://github.com/beyondessential/tupaia-backlog/issues/663 is implemented
-  /**
-   * Find the matching data sources or default to 1:1 mapping with dhis, as only mappings
-   * with non-standard rules are kept in the db
-   *
-   * @param {DataSourceSpec} dataSourceSpec
-   * @returns {Promise<Array>}
-   */
-  async findOrDefault(dataSourceSpec) {
-    assertDataSourceSpecIsValid(dataSourceSpec);
-
-    const records = await this.find(dataSourceSpec);
-    const codeToRecord = keyBy(records, 'code');
-    const { code: codeInput } = dataSourceSpec;
-    const codes = Array.isArray(codeInput) ? codeInput : [codeInput];
-
-    return Promise.all(
-      codes.map(code => codeToRecord[code] || this.getDefault({ ...dataSourceSpec, code })),
+  getDataGroupsThatIncludeElement = async elementConditions => {
+    const dataElement = await this.findOne({ ...elementConditions, type: DATA_ELEMENT });
+    if (!dataElement) {
+      return [];
+    }
+
+    // TODO use `this.find` with joins after
+    // https://github.com/beyondessential/tupaia-backlog/issues/662 is implemented
+    const dataGroups = await this.database.executeSql(
+      `
+        SELECT dg.* FROM data_source dg
+        JOIN data_element_data_group dedg ON dedg.data_group_id = dg.id
+        WHERE dg.type = 'dataGroup' and dedg.data_element_id = ?;
```

Comment:
Ideally we shouldn't use raw SQL except for performance reasons/exotic syntax etc...

I tried to use model queries with `multiJoin`, but it didn't work since I'll be joining the same table (`data_source`) twice. We could certainly update model queries to handle such cases, but I felt this was extending the scope of this task, and  that using a raw query until https://github.com/beyondessential/tupaia-backlog/issues/662 is implemented would be a reasonable compromise.